google/cloud/sql/connector/connector.py

```diff
@@ -238,6 +239,9 @@ async def connect_async(
         # attempt to make connection to Cloud SQL instance
         try:
             instance_data, ip_address = await instance.connect_info(ip_type)
+            # resolve DNS name into IP address for PSC
+            if ip_type.value == "PSC":
+                ip_address = socket.gethostbyname(ip_address)
```

Comment:
Is there an async option for this?

Reply:
@enocom there is no async version of `gethostbyname` however there is an async version of [`getaddrinfo`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.getaddrinfo) which can be used to retrieve the IP address. It is not as clean as you can see that it returns a list and then I am grabbing the first element of the list. It does seem to work though.